src/main/java/pl/cleankod/exchange/core/usecase/FindAccountUseCase.java

```diff
@@ -1,22 +1,44 @@
 package pl.cleankod.exchange.core.usecase;
 
+import java.util.Currency;
+
+import org.apache.commons.lang3.StringUtils;
 import pl.cleankod.exchange.core.domain.Account;
+import pl.cleankod.exchange.core.domain.FindAccountFailedReason;
 import pl.cleankod.exchange.core.gateway.AccountRepository;
-
-import java.util.Optional;
+import pl.cleankod.exchange.entrypoint.exception.NotFoundException;
+import pl.cleankod.util.domain.Result;
 
 public class FindAccountUseCase {
+
     private final AccountRepository accountRepository;
+    private final ConvertCurrencyUseCase convertCurrencyUseCase;
 
-    public FindAccountUseCase(AccountRepository accountRepository) {
+    public FindAccountUseCase(final AccountRepository accountRepository, final ConvertCurrencyUseCase convertCurrencyUseCase) {
         this.accountRepository = accountRepository;
+        this.convertCurrencyUseCase = convertCurrencyUseCase;
+    }
+
+    public Account execute(Account.Id id, String targetCurrency) {
+        final var existingAccount = accountRepository.find(id)
+            .orElseThrow(() -> new NotFoundException("Account does not exist"));
```

Comment:
Throwing an exception here is stepping out of the convention you introduced elsewhere. The other public method has a different one. What is the reason for this incoherence?